packages/api-cli/src/api.ts

```diff
@@ -187,15 +201,30 @@ async function makeTx ({ fn, log }: CallInfo): Promise<void> {
   assert(CRYPTO.includes(sign), `The crypto type can only be one of ${CRYPTO.join(', ')} found '${sign}'`);
 
   const keyring = new Keyring();
-  const account = keyring.createFromUri(seed, {}, sign as 'ed25519');
+  let auth = keyring.createFromUri(seed, {}, sign as 'ed25519');
+
+  let signable;
+  if (sudo) {
+    const provider = new WsProvider(ws);
+    const api = await ApiPromise.create({ provider });
+    const adminId = await api.query.sudo.key();
+    // attempt to get the admin key. This should only be in the keyring if the user
+    // has provided the superuser's seed. We still attempt to fetch the admin ID's
+    // keypair; if that fetch fails, then we can fail immediately instead of waiting
+    // for the tx to be rejected.
+    auth = keyring.getPair(adminId.toString());
+    signable = api.tx.sudo.sudo((fn(...params) as unknown) as Proposal);
+  } else {
+    signable = fn(...params);
+  }
 
-  return fn(...params).signAndSend(account, (result: SubmittableResult): void => {
+  return signable.signAndSend(auth, (result: SubmittableResult): void => {
```

Comment:
As per the comment above, if auth does not match the on-chain sudo, this will fail.